.github/workflows/dbt_slim_ci.yml

```diff
@@ -59,7 +59,7 @@ jobs:
         run: "dbt test --select state:new state:modified --state ."
 
       - name: dbt run
-        run: "dbt run --select state:modified,config.materialized:incremental"
+        run: "dbt run --select state:modified,config.materialized:incremental --state ."
```

Comment:
there's ways to skip this step if it's irrelevant by adding an if statement. https://docs.github.com/en/actions/learn-github-actions/expressions 

you could do something like dbt lst --select config.materialized:incremental in a previous step and set an github arg to false if the output is none. 

tbh we should implement more if statements in this action in general to avoid running steps we don't need. not a bad idea to start here if we want.

Reply:
just learning the hard way you either need it or now that you can skip it 😅

Reply:
adding the ```--state``` flag worked, but happy to dig into that link tomorrow

Reply:
This is an example where we set an env variable if thats helpful.     
      - name: Set environment variables
        run: |
          echo "GIT_SHA=$(echo ${{ github.sha }} | tr - _ | cut -c1-8)" >> $GITHUB_ENV
          
         
Something like this could work. `wc -l` will give you a line count. And you could say if the line count is zero to skip the step. 
          ` dbt ls {your selection logic} | wc -l ` 

      - name: Count incremental models
        run: |
          echo "INC_MODEL_COUNT=$(echo dbt ls --select state:modified,config.materialized:incremental --state .  | wc -l)" >> $GITHUB_ENV

Reply:
> adding the `--state` flag worked, but happy to dig into that link tomorrow

This works but what I'm recommending would make it faster because we wouldn't run a dbt command at all. But it's not super important.

Reply:
without looking closely yet, i think we want to run the dbt command though -- the intention of this addition is to make incremental models run & test twice. unless i'm misunderstanding which piece you mention to not run

Reply:
without looking closely yet, i think we want to run the dbt command though -- the intention of this addition is to make incremental models run & test twice. unless i'm misunderstanding which piece you mention to not run